inflection/src/inflection/grammar/synthesis/SrGrammarSynthesizer_SrDisplayFunction.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 and later: Unicode, Inc. and others.
+ * License & terms of use: http://www.unicode.org/copyright.html
+ */
+#include <inflection/grammar/synthesis/SrGrammarSynthesizer_SrDisplayFunction.hpp>
+
+#include <inflection/tokenizer/TokenChain.hpp>
+#include <inflection/tokenizer/Tokenizer.hpp>
+#include <inflection/tokenizer/TokenizerFactory.hpp>
+#include <inflection/dialog/SemanticFeature.hpp>
+#include <inflection/dialog/SemanticFeatureModel.hpp>
+#include <inflection/dialog/SemanticFeatureModel_DisplayData.hpp>
+#include <inflection/dialog/DisplayValue.hpp>
+#include <inflection/grammar/synthesis/GrammemeConstants.hpp>
+#include <inflection/grammar/synthesis/GrammarSynthesizerUtil.hpp>
+#include <inflection/lang/StringFilterUtil.hpp>
+#include <inflection/util/LocaleUtils.hpp>
+#include <inflection/util/UnicodeSetUtils.hpp>
+#include <inflection/npc.hpp>
+#include <iterator>
+#include <memory>
+
+namespace inflection::grammar::synthesis {
+
+SrGrammarSynthesizer_SrDisplayFunction::SrGrammarSynthesizer_SrDisplayFunction(const ::inflection::dialog::SemanticFeatureModel& model)
+    : super()
+    , dictionary(*npc(::inflection::dictionary::DictionaryMetaData::createDictionary(::inflection::util::LocaleUtils::SERBIAN())))
+    , countFeature(model.getFeature(GrammemeConstants::NUMBER))
+    , genderFeature(model.getFeature(GrammemeConstants::GENDER))
+    , partOfSpeechFeature(model.getFeature(GrammemeConstants::POS))
+    , tokenizer(::inflection::tokenizer::TokenizerFactory::createTokenizer(::inflection::util::LocaleUtils::SERBIAN()))
+    , dictionaryInflector(::inflection::util::LocaleUtils::SERBIAN(),{
+            {GrammemeConstants::POS_NOUN(), GrammemeConstants::POS_ADJECTIVE(), GrammemeConstants::POS_VERB()},
+            {GrammemeConstants::PERSON_THIRD(), GrammemeConstants::PERSON_FIRST(), GrammemeConstants::PERSON_SECOND()},
+            {GrammemeConstants::NUMBER_SINGULAR(), GrammemeConstants::NUMBER_PLURAL()},
+            {GrammemeConstants::GENDER_MASCULINE(), GrammemeConstants::GENDER_FEMININE(), GrammemeConstants::GENDER_NEUTER()}
+    }, {}, true)
+{
+}
+
+SrGrammarSynthesizer_SrDisplayFunction::~SrGrammarSynthesizer_SrDisplayFunction()
+{
+}
+
+::inflection::dialog::DisplayValue * SrGrammarSynthesizer_SrDisplayFunction::getDisplayValue(const dialog::SemanticFeatureModel_DisplayData &displayData, const ::std::map<::inflection::dialog::SemanticFeature, ::std::u16string> &constraints, bool enableInflectionGuess) const
+{
+    const auto displayValue = GrammarSynthesizerUtil::getTheBestDisplayValue(displayData, constraints);
+    if (displayValue == nullptr) {
+        return nullptr;
+    }
+    ::std::u16string displayString = displayValue->getDisplayString();
+    if (displayString.empty()) {
+        return nullptr;
+    }
+
+    // To make compiler quiet about unused variable.
+    if (enableInflectionGuess)
+        return nullptr;
+
+    // TODO Implement the rest
```

Comment:
Do you have plans to implement this TODO?

Reply:
I still have some tests failing. My goal is to have tests pass, submit this PR, then open a new one with affecting only this file.
WDYT?

Reply:
That sounds like a good plan.